src/js/views/patients/worklist/worklist_views.js

```diff
@@ -212,6 +213,26 @@ const ListView = CollectionView.extend({
         return matcher.test(searchString);
       });
     });
+
+    this.state.set('lastSelectedId', null);
+  },
+  onClickShiftMultiSelect(selectedModelId) {
+    const lastSelectedId = this.state.get('lastSelectedId');
+
+    let selectedCollection = this.collection.clone();
+    selectedCollection.reset(this.children.map('model'));
+
+    const selectedIndex = selectedCollection.findIndex(model => model.id === selectedModelId);
+    const lastSelectedIndex = selectedCollection.findIndex(model => model.id === lastSelectedId);
+
+    const minIndex = Math.min(selectedIndex, lastSelectedIndex);
+    const maxIndex = Math.max(selectedIndex, lastSelectedIndex);
+
+    selectedCollection = selectedCollection.filter((model, index) => {
```

Comment:
actually I think this needs to be `selectedCollection.reset` and keep it a `const` `filter` returns an array of models and not a collection.. which works in this instance because of `Array.prototype.reduce`, but it's not a collection.
Also I think you were using `Array.prototype.slice` previously, and I'm just now realizing we can do that here too `selectedCollection.slice`